ndcube/ndcollection.py

```diff
@@ -0,0 +1,262 @@
+        collection_items = [[slice(None)] * len(self[key].dimensions) for key in self]
+        # Define empty list to hold aligned axes dropped by the slicing.
+        drop_aligned_axes_indices = []
+
+        # Case 1: int
+        # First aligned axis is dropped.
+        if isinstance(item, int):
+            drop_aligned_axes_indices = [0]
+            # Insert item to each cube's slice item.
+            for i, key in enumerate(self):
+                collection_items[i][self.aligned_axes[key][0]] = item
+
+        # Case 2: slice
+        # If only one element in interval of slice, first aligned axis is dropped.
+        elif isinstance(item, slice):
+            # Insert item to each cube's slice item.
+            for i, key in enumerate(self):
+                collection_items[i][self.aligned_axes[key][0]] = item
+            # Note that slice interval's of 1 result in an axis of length 1.
+            # The axis is not dropped.
+
+        # Case 3: tuple of ints/slices
+        # Search sub-items within tuple for ints or 1-interval slices.
+        elif isinstance(item, tuple):
+            # Ensure item is not longer than number of aligned axes
+            if len(item) > self.n_aligned_axes:
+                raise IndexError("Too many indices")
+            # If item is tuple, search sub-items for ints or 1-interval slices.
+            for i, axis_item in enumerate(item):
+                if isinstance(axis_item, int):
+                    drop_aligned_axes_indices.append(i)
+                for j, key in enumerate(self):
+                    collection_items[j][self.aligned_axes[key][i]] = axis_item
+
+        else:
+            raise TypeError("Unsupported slicing type: {0}".format(axis_item))
+
+        # Use indices of dropped axes determine above to update aligned_axes
+        # by removing any that have been dropped.
+        drop_aligned_axes_indices = np.array(drop_aligned_axes_indices)
+        new_aligned_axes = collection_utils._update_aligned_axes(
+                drop_aligned_axes_indices, self.aligned_axes, self._first_key)
+
+        return collection_items, new_aligned_axes
+
+    def copy(self):
+        return copy.deepcopy(self)
+
+    def setdefault(self):
+        raise NotImplementedError("NDCollection does not support setdefault.")
+
+    def popitem(self):
+        raise NotImplementedError("NDCollection does not support popitem.")
+
+    def pop(self, key):
+        """Removes the cube corresponding to the key from the collection and returns it."""
+        # Extract desired cube from collection.
+        popped_cube = super().pop(key)
+        # Delete corresponding aligned axes
+        popped_aligned_axes = self.aligned_axes.pop(key)
+        # If first key removed, update.
+        if key == self._first_key:
+            self._first_key = list(self.keys())[0]
+
+        return popped_cube
+
+    def update(self, key, data, aligned_axes):
```

Comment:
I am not a big fan about how this method deviates from the `dict.update` method. Should it not just take in another instance of `NDCollection`? like `dict.update` takes a `dict`? We could add an extend method which takes these three arguments.